src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests2.cs

```diff
@@ -0,0 +1,311 @@
+                        N(SyntaxKind.CloseBraceToken);
+                    }
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void ExtendedPropertySubpattern_04()
+        {
+            UsingExpression(@"e is { name[0]: p }",
+                // (1,15): error CS1003: Syntax error, ',' expected
+                // e is { name[0]: p }
+                Diagnostic(ErrorCode.ERR_SyntaxError, ":").WithArguments(",", ":").WithLocation(1, 15),
+                // (1,17): error CS1003: Syntax error, ',' expected
+                // e is { name[0]: p }
+                Diagnostic(ErrorCode.ERR_SyntaxError, "p").WithArguments(",", "").WithLocation(1, 17));
+
+
+            N(SyntaxKind.IsPatternExpression);
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "e");
+                }
+                N(SyntaxKind.IsKeyword);
+                N(SyntaxKind.RecursivePattern);
+                {
+                    N(SyntaxKind.PropertyPatternClause);
+                    {
+                        N(SyntaxKind.OpenBraceToken);
+                        N(SyntaxKind.Subpattern);
+                        {
+                            N(SyntaxKind.TypePattern);
+                            {
+                                N(SyntaxKind.ArrayType);
+                                {
+                                    N(SyntaxKind.IdentifierName);
+                                    {
+                                        N(SyntaxKind.IdentifierToken, "name");
+                                    }
+                                    N(SyntaxKind.ArrayRankSpecifier);
+                                    {
+                                        N(SyntaxKind.OpenBracketToken);
+                                        N(SyntaxKind.NumericLiteralExpression);
+                                        {
+                                            N(SyntaxKind.NumericLiteralToken, "0");
+                                        }
+                                        N(SyntaxKind.CloseBracketToken);
+                                    }
+                                }
+                            }
+                        }
+                        M(SyntaxKind.CommaToken);
+                        N(SyntaxKind.Subpattern);
+                        {
+                            N(SyntaxKind.ConstantPattern);
+                            {
+                                N(SyntaxKind.IdentifierName);
+                                {
+                                    N(SyntaxKind.IdentifierToken, "p");
+                                }
+                            }
+                        }
+                        N(SyntaxKind.CloseBraceToken);
+                    }
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void ExtendedPropertySubpattern_05()
```

Comment:
Consider also testing a pointer access syntax

Reply:
Unfortunately that wouldn't give useful diagnostics. While patterns and expression are only partially subset of each other.

Any ideas?

Reply:
I'd expect ParsePattern  cover those expressions, probably need some adjustment. I'll look if we can do a better job here.

Reply:
As long as we have errors, I think it's fine. I see this is now covered by `ExtendedPropertyPatterns_04` which looks fine.